src/model/MultiValueColumn.ts

```diff
@@ -0,0 +1,281 @@
+/**
+ * Created by bikramkawan on 24/11/2016.
+ */
+import * as d3 from 'd3';
+import ValueColumn from './ValueColumn';
+
+
+export class CustomSortCalculation {
+
+  constructor(private a_val: number[], private b_val: number []) {
+    this.b_val = b_val;
+    this.a_val = a_val;
+  }
+
+
+  sum() {
+    return (d3.sum(this.a_val) - d3.sum(this.b_val));
+  }
+
+  min() {
+    return (d3.min(this.a_val) - d3.min(this.b_val));
+
+  }
+
+
+  max() {
+    return (d3.max(this.a_val) - d3.max(this.b_val));
+  }
+
+  mean() {
+
+    return (d3.mean(this.a_val) - d3.mean(this.b_val));
+  }
+
+  median() {
+    return (d3.median(this.a_val)) - (d3.median(this.b_val));
+
+  }
+
+  q1() {
+
+    return (d3.quantile(this.a_val, 0.25)) - (d3.quantile(this.b_val, 0.25));
+
+  }
+
+  q3() {
+
+    return (d3.quantile(this.a_val, 0.75)) - (d3.quantile(this.b_val, 0.75));
+
+  }
+}
+
+// function getPercentile(data, percentile) {
+//
+//       var index = (percentile / 100) * data.length;
+//       var result;
+//       if (Math.floor(index) === index) {
+//         result = (data[(index - 1)] + data[index]) / 2;
+//       } else {
+//         result = data[Math.floor(index)];
+//       }
+//       return result;
+//     }
+
+function numSort(a, b) {
+  return a - b;
+}
+
+
+export default class MultiValueColumn extends ValueColumn<number[] > {
+  private sortBy;
```

Comment:
current sortCriteria should be also dumped and restored to reproduce the full state